mlflow/projects/submitted_run.py

```diff
@@ -46,6 +46,11 @@ def cancel(self):
         """
         pass
 
+    @property
```

Comment:
It unfortunately seems this doesn't quite work as expected in Python 2.7, in that subclasses aren't required to define `run_id` (see this [SO post](https://stackoverflow.com/questions/5960337/how-to-create-abstract-properties-in-python-abstract-classes)). I also can't seem to find anything in `six` that'd make this just work for Python 2 & 3.

The current implementation is fine with me, since we'll achieve the desired behavior in Python 3 which is strictly better than the current situation (maybe we can just comment that this doesn't actually make `run_id` abstract in Python 2)

Reply:
That's fine if it's not actually abstract in Python 2, since we cross-compile.